### ISSUE TITLE
--Level Auto doesn't detect level properly

### ISSUE THREAD

**Issue by Eldenissio** [rigaya/NVEnc#316]
Hi.

I have a movie I transcode with this powershell line:
nvencc64.exe --codec hevc --avhw --input-analyze 60 --avsync cfr --output-thread 1 --video-track 1 --cqp 21 --preset P7 --output-depth 10 --lookahead 24 --strict-gop --profile main10 --tier high --level auto --aq --mv-precision q-pel --colormatrix auto --colorprim auto --transfer auto --colorrange auto --chromaloc auto --master-display copy --max-cll copy  --cuda-schedule spin --audio-copy 1,2 --chapter-copy --sub-copy 2,4  -i "G:\For_transcode\Mulan.2020.mkv" -o "G:\Transcoded\2020 - Mulan.mkv"

the original file has:

Video
ID                             : 1
Format                         : HEVC
Format/Info                    : High Efficiency Video Coding
Format profile                 **: Main 10@L5.1@High**
Codec ID                       : V_MPEGH/ISO/HEVC
Duration                       : 1 h 55 min
Bit rate                       : 54.8 Mb/s
Width                          : 3 840 pixels
Height                         : 2 160 pixels
Display aspect ratio           : 16:9
Frame rate mode                : Constant
Frame rate                     : 23.976 (24000/1001) FPS
Color space                    : YUV
Chroma subsampling             : 4:2:0 (Type 2)
Bit depth                      : 10 bits
Bits/(Pixel*Frame)             : 0.275
Stream size                    : 44.1 GiB (91%)
Writing library                : ATEME Titan File 3.9.6 (4.9.6.2)        
Language                       : English
Default                        : Yes
Forced                         : No
Color range                    : Limited
Color primaries                : BT.2020
Transfer characteristics       : SMPTE ST 2084
Matrix coefficients            : BT.2020 non-constant
Mastering display color primar : R: x=0.680000 y=0.320000, G: x=0.265000 y=0.690000, B: x=0.150000 y=0.060000, White point: x=0.312700 y=0.329000
Mastering display luminance    : min: 0.0001 cd/m2, max: 1000.0000 cd/m2

When Transcode starts, I get a resulting file:

Video
ID                             : 1
Format                         : HEVC
Format/Info                    : High Efficiency Video Coding
Format profile                 : **Main 10@L5@Main**
Codec ID                       : V_MPEGH/ISO/HEVC
Duration                       : 1 h 55 min
Width                          : 3 840 pixels
Height                         : 2 160 pixels
Display aspect ratio           : 16:9
Frame rate mode                : Constant
Frame rate                     : 23.976 (24000/1001) FPS
Color space                    : YUV
Chroma subsampling             : 4:2:0 (Type 2)
Bit depth                      : 10 bits
Default                        : Yes
Forced                         : No
Color range                    : Limited
Color primaries                : BT.2020
Transfer characteristics       : SMPTE ST 2084
Matrix coefficients            : BT.2020 non-constant
DURATION                       : 01:55:11.238000000
Mastering display color primar : R: x=0.680000 y=0.320000, G: x=0.265000 y=0.690000, B: x=0.150000 y=0.060000, White point: x=0.312700 y=0.329000
Mastering display luminance    : min: 0.0001 cd/m2, max: 1000.0000 cd/m2

Basically the result has Level 5 instead of 5.1
I'm not sure if it's the reason for colours to get a bit over-brightened so they loose some saturation, but I've noticed this in version  5.29. I've updated just now from 5.18 which had no colour difference with the same command in powershell.

I will do a test run with manual Level selection to see if it's the cause of the issue for colour shift, but Level is definitely wrong

**Comment by rigaya** [comment]
"--level auto" is **not** a option to copy level from the input file.

"--level auto" means to let the level determined by the NVIDIA driver, depending on the output resolution, framerate, and other conditions, so this is a expected behavior. 3840x2160 24fps can be supported by HEVC Level5 without problem.
